http/http-api/src/main/java/com/okta/commons/http/QueryString.java

```diff
@@ -35,18 +36,29 @@ public QueryString(){}
 
     public QueryString(Map<String,?> source) {
         super();
+
         if (!Collections.isEmpty(source)) {
-            for(Map.Entry<String,?> entry : source.entrySet()) {
+            for (Map.Entry<String,?> entry : source.entrySet()) {
                 String key = entry.getKey();
                 Object value = entry.getValue();
-                String sValue = value != null ? String.valueOf(value) : null;
+                String sValue = null;
+                if (value != null) {
+                    if (value instanceof Date) {
+                        sValue = RequestUtils.getFormattedDate((Date) value);
+                    } else {
+                        sValue = String.valueOf(value);
+                    }
+                }
```

Comment:
nit: move the duplicate to string logic to a single method and reuse it in `put` below

Reply:
Done